File: Data Structures & Algorithms/missing-number/submission-0.java
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int res = (n * (n+1)) / 2;
        for(int i = 0; i < n; i++){
            res -= nums[i];
        }
        return res;
    }
}

File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        res = []
        nums.sort()
        def helper(i, subset):
            if i >= len(nums):
                res.append(subset.copy())
                return

            subset.append(nums[i])
            helper(i+1, subset)
            subset.pop()
            while i < len(nums)-1 and nums[i] == nums[i+1]:
                i += 1
            helper(i+1, subset)
            
        
        helper(0, [])
        return res